update and codify dependencies

grundrauschen opened the pull request.

grundrauschen committed:
update and codify versions of dependencies

grundrauschen committed:
update CHANGELOG

grundrauschen committed:
fix some build errors

mgumz reviewed (commented):
what about just using the binary? https://github.com/kelseyhightower/confd/releases

grundrauschen committed:
Use binary version of `confd` instead of compiling
The version of `confd` will be checked against a given checksum

grundrauschen committed:
add additional labels following label-schema

grundrauschen committed:
add mention of change to use binary confd in changelog

grundrauschen committed:
mention of security fix due to new build of container image

grundrauschen committed:
add feature to build images with frozen versions

`MANIFEST` was added to pin the versions used in the created images.
Further the Dockerfiles was updated to use the exact versions of this
file.
`freeze_apk_versions` was added update the MANIFEST file after updating
the software.

grundrauschen committed:
add table of content to README and add corresponding Makefile
